pkg/controller/crdmirroring/types/interface.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package types
+
+import (
+	"k8s.io/client-go/tools/cache"
+)
+
+type GenericInformer interface {
+	Informer() cache.SharedIndexInformer
```

Comment:
The consumers can use `cache.SharedIndexInformer` directly

Reply:
If we use this, we can't recover lister from handler.

Reply:
As commented in https://github.com/vmware-tanzu/antrea/pull/1799#discussion_r597565709:
The controller takes care of generic logic and is resource agnostic, so it just needs to register generic handlers to generic informer. 
A concrete handler takes care of a specific kind of resources, it's the only one that needs the Lister. I think this would work:
```
networkPolicyMirroringHandler := crdhandler.NewNetworkPolicyHandler(anpInformer.Lister(), legacyAnpInformer.Lister(), client, legacyClient, "AntreaNetworkPolicy")
networkPolicyMirroringController = crdmirroring.NewController(anpInformer.Informer(), legacyAnpInformer.Informer(), networkPolicyMirroringHandler)

Reply:
Good idea. Corrected as your suggestion.